### DISCUSSION TITLE
Auto-running the extension at some time intervals

### DISCUSSION THREAD

**Discussion by ssbarnea** [seachicken/gh-poi#117] [Ideas]
I have >100 repositories I work with and I would find very useful if this extension could be configured to run automatically every now and they to do the cleaning.

I guess that a good way to do it is if we could make it hook it to run on other `gh` operations that one would usually do, such `gh create` or `gh checkout`.

**Comment by seachicken** [comment]
Thank you for the suggestion! It is useful to run poi across multiple repositories easily.
Since I have aimed to provide as simple a function as possible, I am not sure whether I should aim to support this function in poi.
I came up with the following, is there anything that might help?
- Run for each repository with an alias (e.g., `alias ghco='gh poi && gh pr checkout'`)
- Run in background with crontab
- Run all at once with sh:
  ```sh
  #!/bin/bash

  repos=("/Users/seito/git/GitHub/gh-poi" "/Users/seito/git/GitHub/inga")

  for repo in "${repos[@]}"; do
    (cd "$repo" && gh poi)
  done
  ```
For the last two, we need to be careful about the [rate limit](https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28) if there are many repositories.
I would like to know more about the demand for this feature.
